### PR TITLE
Warhammer 10 cards + 1 tokens

### PR DESCRIPTION
**Pull request by Simisays**

### CODE REVIEW

File: forge-gui/res/cardsfolder/upcoming/genestealer_locus.txt [hunk 1]
```diff
@@ -0,0 +1,9 @@
+Name:Genestealer Locus
+ManaCost:3 U
+Types:Creature Tyranid Human
+PT:3/3
+T:Mode$ Attacks | ValidCard$ Creature | Attacked$ You | TriggerZones$ Battlefield | Execute$ TrigDebuff | TriggerDescription$ Neurotraumal Rod — Whenever a creature attacks you, it gets -1/-0 until end of turn.
+SVar:TrigDebuff:DB$ Pump | Defined$ TriggeredAttacker | NumAtt$ -1
+T:Mode$ Attacks | ValidCard$ Creature | Attacked$ Player.Opponent,Planeswalker.OppCtrl | TriggerZones$ Battlefield | Execute$ TrigPump | TriggerDescription$ Whenever a creature attacks one of your opponents or a planeswalker an opponent controls, that creature gets +0/+1 until end of turn.
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/genestealer_locus.txt:7]: doesn't match oracle

File: forge-gui/res/cardsfolder/upcoming/genestealer_patriarch.txt [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+SVar:TrigInfectionCounter:DB$ PutCounter | CounterType$ INFECTION | CounterNum$ 1 | ValidTgts$ Creature.ControlledBy TriggeredDefendingPlayer |  TgtPrompt$ Put an infection counter on target creature defending player controls. 
+T:Mode$ ChangesZone | Origin$ Battlefield | Destination$ Graveyard | ValidCard$ Creature.counters_GE1_INFECTION | Execute$ DBCopy | TriggerDescription$ Children of the Cult — Whenever a creature with an infection counter on it dies, you create a token that's a copy of that creature, except it's a Tyranid in addition to its other types.
+SVar:DBCopy:DB$ CopyPermanent | Defined$ TriggeredCardLKICopy | AddTypes$ Tyranid 
+DeckHas:Type$Counters|Token
+Oracle:Genestealer’s Kiss — Whenever Genestealer Patriarch attacks, put an infection counter on target creature defending player controls.\nChildren of the Cult — Whenever a creature with an infection counter on it dies, you create a token that's a copy of that creature, except it's a Tyranid in addition to its other types.
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/genestealer_patriarch.txt:10]: curly apost

File: forge-gui/res/cardsfolder/upcoming/genestealer_patriarch.txt [hunk 3]
```diff
@@ -0,0 +1,10 @@
+Name:Genestealer Patriarch
+ManaCost:3 U
+Types:Creature Tyranid Human
+PT:4/3
+T:Mode$ Attacks | ValidCard$ Card.Self | Execute$ TrigInfectionCounter | TriggerDescription$ Genestealer’s Kiss — Whenever CARDNAME attacks, put an infection counter on target creature defending player controls.
+SVar:TrigInfectionCounter:DB$ PutCounter | CounterType$ INFECTION | CounterNum$ 1 | ValidTgts$ Creature.ControlledBy TriggeredDefendingPlayer |  TgtPrompt$ Put an infection counter on target creature defending player controls. 
+T:Mode$ ChangesZone | Origin$ Battlefield | Destination$ Graveyard | ValidCard$ Creature.counters_GE1_INFECTION | Execute$ DBCopy | TriggerDescription$ Children of the Cult — Whenever a creature with an infection counter on it dies, you create a token that's a copy of that creature, except it's a Tyranid in addition to its other types.
+SVar:DBCopy:DB$ CopyPermanent | Defined$ TriggeredCardLKICopy | AddTypes$ Tyranid 
+DeckHas:Type$Counters|Token
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/genestealer_patriarch.txt:9]: these are not Types

File: forge-gui/res/cardsfolder/upcoming/genestealer_patriarch.txt [hunk 4]
```diff
@@ -0,0 +1,10 @@
+Name:Genestealer Patriarch
+ManaCost:3 U
+Types:Creature Tyranid Human
+PT:4/3
+T:Mode$ Attacks | ValidCard$ Card.Self | Execute$ TrigInfectionCounter | TriggerDescription$ Genestealer’s Kiss — Whenever CARDNAME attacks, put an infection counter on target creature defending player controls.
+SVar:TrigInfectionCounter:DB$ PutCounter | CounterType$ INFECTION | CounterNum$ 1 | ValidTgts$ Creature.ControlledBy TriggeredDefendingPlayer |  TgtPrompt$ Put an infection counter on target creature defending player controls.
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/genestealer_patriarch.txt:6]: spacing

File: forge-gui/res/cardsfolder/upcoming/genestealer_patriarch.txt [hunk 5]
```diff
@@ -0,0 +1,10 @@
+Name:Genestealer Patriarch
+ManaCost:3 U
+Types:Creature Tyranid Human
+PT:4/3
+T:Mode$ Attacks | ValidCard$ Card.Self | Execute$ TrigInfectionCounter | TriggerDescription$ Genestealer’s Kiss — Whenever CARDNAME attacks, put an infection counter on target creature defending player controls.
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/genestealer_patriarch.txt:5]: curly apost

File: forge-gui/res/cardsfolder/upcoming/haruspex.txt [hunk 6]
```diff
@@ -0,0 +1,11 @@
+Name:Haruspex
+ManaCost:3 G
+Types:Creature Tyranid
+PT:2/2
+T:Mode$ ChangesZone | Origin$ Battlefield | Destination$ Graveyard | ValidCard$ Creature | TriggerZones$ Battlefield | Execute$ TrigPutCounter | TriggerDescription$ Rapacious Hunger — Whenever another creature dies, put a +1/+1 counter on CARDNAME.
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/haruspex.txt:5]: Shouldn't trigger for itself

File: forge-gui/res/cardsfolder/upcoming/genestealer_patriarch.txt (hunk 4 above)
Comment (Northmoc) [reply forge-gui/res/cardsfolder/upcoming/genestealer_patriarch.txt:6]: usually TgtPrompt doesn't have `.`

File: forge-gui/res/cardsfolder/upcoming/genestealer_patriarch.txt (hunk 3 above)
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/genestealer_patriarch.txt:9]: SVar:HasAttackEffect:TRUE

File: forge-gui/res/cardsfolder/upcoming/haruspex.txt [hunk 7]
```diff
@@ -0,0 +1,11 @@
+Name:Haruspex
+ManaCost:3 G
+Types:Creature Tyranid
+PT:2/2
+T:Mode$ ChangesZone | Origin$ Battlefield | Destination$ Graveyard | ValidCard$ Creature | TriggerZones$ Battlefield | Execute$ TrigPutCounter | TriggerDescription$ Rapacious Hunger — Whenever another creature dies, put a +1/+1 counter on CARDNAME.
+SVar:TrigPutCounter:DB$ PutCounter | Defined$ Self | CounterType$ P1P1 | CounterNum$ 1
+A:AB$ Mana | Cost$ T SubCounter<X/P1P1> | Produced$ Any  | PreCostDesc$ Devouring Monster - | Amount$ X | SpellDescription$ Add X mana of any one color.
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/haruspex.txt:7]: spacing, also capping for PrecostDesc

File: forge-gui/res/cardsfolder/upcoming/nexos.txt [hunk 8]
```diff
@@ -0,0 +1,7 @@
+Name:Nexos
+ManaCost:1 G
+Types:Creature Human Tyranid Advisor
+PT:2/2
+S:Mode$ Continuous | Affected$ Land.Basic+YouCtrl | AddAbility$ Mana | Description$ Strategic Coordinator — Basic lands you control have "{T}: Add {C}{C}. Spend this mana only on costs that contain {X}."
+SVar:Mana:AB$ Mana | Cost$ T | Produced$ Any | Produced$ C | Amount$ 2 | RestrictValid$ CostContainsX | SpellDescription$ Add {C}{C}. Spend this mana only on costs that contain {X}.
+Oracle:Strategic Coordinator — Basic lands you control have "{T}: Add {C}{C}. Spend this mana only on costs that contain {X}."
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/nexos.txt:7]: AI:RemoveDeck:Random

File: forge-gui/res/cardsfolder/upcoming/old_one_eye.txt [hunk 9]
```diff
@@ -0,0 +1,13 @@
+Name:Old One Eye
+ManaCost:5 G
+Types:Legendary Creature Tyranid
+PT:6/6
+K:Trample
+S:Mode$ Continuous | Affected$ Creature.Other+YouCtrl | AddKeyword$ Trample | Description$ Other creatures you control have trample.
+T:Mode$ ChangesZone | Origin$ Any | Destination$ Battlefield | ValidCard$ Card.Self | Execute$ TrigToken | TriggerDescription$ When CARDNAME enters the battlefield, create a 5/5 green Tyranid creature token.
+SVar:TrigToken:DB$ Token | TokenAmount$ 1 | TokenScript$ g_5_5_tyranid | TokenOwner$ You
+T:Mode$ Phase | PreCombatMain$ True | ValidPlayer$ You |  ValidPlayer$ You | TriggerZones$ Graveyard | PresentZone$ Graveyard | PresentPlayer$ You | Execute$ TrigReturn | TriggerDescription$ Fast Healing — At the beginning of your precombat main phase, you may discard two cards. If you do, return CARDNAME from your graveyard to your hand.
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/old_one_eye.txt:9]: `ValidPlayer$ You |  ValidPlayer$ You`

Comment (Northmoc) [reply forge-gui/res/cardsfolder/upcoming/old_one_eye.txt:9]: Missing `IsPresent$ Card.StrictlySelf`

File: forge-gui/res/cardsfolder/upcoming/old_one_eye.txt [hunk 10]
```diff
@@ -0,0 +1,13 @@
+Name:Old One Eye
+ManaCost:5 G
+Types:Legendary Creature Tyranid
+PT:6/6
+K:Trample
+S:Mode$ Continuous | Affected$ Creature.Other+YouCtrl | AddKeyword$ Trample | Description$ Other creatures you control have trample.
+T:Mode$ ChangesZone | Origin$ Any | Destination$ Battlefield | ValidCard$ Card.Self | Execute$ TrigToken | TriggerDescription$ When CARDNAME enters the battlefield, create a 5/5 green Tyranid creature token.
+SVar:TrigToken:DB$ Token | TokenAmount$ 1 | TokenScript$ g_5_5_tyranid | TokenOwner$ You
+T:Mode$ Phase | PreCombatMain$ True | ValidPlayer$ You |  ValidPlayer$ You | TriggerZones$ Graveyard | PresentZone$ Graveyard | PresentPlayer$ You | Execute$ TrigReturn | TriggerDescription$ Fast Healing — At the beginning of your precombat main phase, you may discard two cards. If you do, return CARDNAME from your graveyard to your hand.
+SVar:TrigReturn:AB$ ChangeZone | Cost$ Discard<2/Card>  | Defined$ Self | Origin$ Graveyard | Destination$ Hand
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/old_one_eye.txt:10]: spacing

File: forge-gui/res/cardsfolder/upcoming/old_one_eye.txt (hunk 9 above)
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/old_one_eye.txt:9]: ValidPlayer$ twice

File: forge-gui/res/cardsfolder/upcoming/the_red_terror.txt [hunk 11]
```diff
@@ -0,0 +1,8 @@
+Name:The Red Terror
+ManaCost:3 R
+Types:Legendary Creature Tyranid
+PT:4/3
+T:Mode$ DamageDone | ValidSource$ Card.RedSource+YouCtrl | ValidTarget$ Player,Permanent | Execute$ TrigPutCounter | TriggerDescription$ Advanced Species — Whenever a red source you control deals damage to one or more permanents and/or players, put a +1/+1 counter on CARDNAME.
+SVar:TrigPutCounter:DB$ PutCounter | Defined$ Self | CounterType$ P1P1 | CounterNum$ 1
+DeckHints:Color$Red
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/the_red_terror.txt:7]: Not needed... since this is a red card, stands to reason the deck will have red in it

Comment (Simisays) [reply forge-gui/res/cardsfolder/upcoming/the_red_terror.txt:7]: @Northmoc Was thinking it is better in mono red but I guess there's no hint for that

File: forge-gui/res/cardsfolder/upcoming/the_red_terror.txt [hunk 12]
```diff
@@ -0,0 +1,8 @@
+Name:The Red Terror
+ManaCost:3 R
+Types:Legendary Creature Tyranid
+PT:4/3
+T:Mode$ DamageDone | ValidSource$ Card.RedSource+YouCtrl | ValidTarget$ Player,Permanent | Execute$ TrigPutCounter | TriggerDescription$ Advanced Species — Whenever a red source you control deals damage to one or more permanents and/or players, put a +1/+1 counter on CARDNAME.
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/the_red_terror.txt:5]: This is a `DamageDealtOnce`

Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/the_red_terror.txt:5]: I think this should be `DamageDealtOnce`

File: forge-gui/res/cardsfolder/upcoming/genestealer_patriarch.txt [hunk 13]
```diff
@@ -0,0 +1,11 @@
+Name:Genestealer Patriarch
+ManaCost:3 U
+Types:Creature Tyranid Human
+PT:4/3
+T:Mode$ Attacks | ValidCard$ Card.Self | Execute$ TrigInfectionCounter | TriggerDescription$ Genestealer's Kiss — Whenever CARDNAME attacks, put an infection counter on target creature defending player controls.
+SVar:TrigInfectionCounter:DB$ PutCounter | CounterType$ INFECTION | CounterNum$ 1 | ValidTgts$ Creature.ControlledBy TriggeredDefendingPlayer | TgtPrompt$ Put an infection counter on target creature defending player controls
+T:Mode$ ChangesZone | Origin$ Battlefield | Destination$ Graveyard | ValidCard$ Creature.counters_GE1_INFECTION | Execute$ DBCopy | TriggerDescription$ Children of the Cult — Whenever a creature with an infection counter on it dies, you create a token that's a copy of that creature, except it's a Tyranid in addition to its other types.
+SVar:DBCopy:DB$ CopyPermanent | Defined$ TriggeredCardLKICopy | AddTypes$ Tyranid 
+DeckHas:Ability$Counters|Token
+SVar:HasAttackEffect:TRUE
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/genestealer_patriarch.txt:10]: here's a spicy one:
DeckHints:Name$Diseased Vermin

**Review by tool4ever — APPROVED** [review]

File: forge-gui/res/cardsfolder/upcoming/the_red_terror.txt [hunk 14]
```diff
@@ -0,0 +1,7 @@
+Name:The Red Terror
+ManaCost:3 R
+Types:Legendary Creature Tyranid
+PT:4/3
+T:Mode$ DamageDealtOnce | ValidSource$ Card.RedSource+YouCtrl | ValidTarget$ Player,Permanent | Execute$ TrigPutCounter | TriggerDescription$ Advanced Species — Whenever a red source you control deals damage to one or more permanents and/or players, put a +1/+1 counter on CARDNAME.
+SVar:TrigPutCounter:DB$ PutCounter | Defined$ Self | CounterType$ P1P1 | CounterNum$ 1
+Oracle:Advanced Species — Whenever a red source you control deals damage to one or more permanents and/or players, put a +1/+1 counter on The Red Terror.
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/the_red_terror.txt:7]: DeckHas:Ability$Counters

**Pull request merged**